# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/win/RemoteAccessible.cpp
# Commit: c2233b3830ef
# Full Hash: c2233b3830efd3e475a76443d22c07e73cda476a
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2022-06-11 09:51:55
# Regressor Bug: 1758689
# File Overlap Count: 1
# Description:
#   Bug 1758689: Create and maintain a viewport cache for hittesting r=Jamie,emilio
#   
#   This hittesting approach is borrowed from our Android implementation.
#   We maintain a cache of ID's for accessibles which are visible in the viewport.
#   This cache is created with a call to `nsLayoutUtils::GetFramesForArea`, which
# ==============================================================================

diff -r 7a17841a34cc -r c2233b3830ef accessible/ipc/win/RemoteAccessible.cpp
--- a/accessible/ipc/win/RemoteAccessible.cpp	Fri Jun 10 23:49:24 2022 +0300
+++ b/accessible/ipc/win/RemoteAccessible.cpp	Fri Jun 10 21:10:18 2022 +0000
@@ -820,6 +820,11 @@
 
 Accessible* RemoteAccessible::ChildAtPoint(
     int32_t aX, int32_t aY, Accessible::EWhichChildAtPoint aWhichChild) {
+  if (StaticPrefs::accessibility_cache_enabled_AtStartup()) {
+    return RemoteAccessibleBase<RemoteAccessible>::ChildAtPoint(aX, aY,
+                                                                aWhichChild);
+  }
+
   RefPtr<IAccessible2_2> target = QueryInterface<IAccessible2_2>(this);
   if (!target) {
     return nullptr;